starfield.js

```diff
@@ -5,97 +5,130 @@
+    var y = Math.sin(angle);
+    var velocity =
+      Math.random() * (this.maxVelocity - this.minVelocity) + this.minVelocity;
+    stars[i] = new Star(x, y, Math.random() * 3 + 1, velocity);
+  }
+  this.stars = stars;
+
+  var self = this;
+  //	Start the timer.
+  this.intervalId = setInterval(function () {
+    self.update();
+    self.draw();
+  }, 1000 / this.fps);
+
+  // Skip 10000 frames so stars start in random places.
+  for (var i = 0; i < 10000; i++) {
+    self.update();
+  }
 };
 
-Starfield.prototype.stop = function() {
-	clearInterval(this.intervalId);
+Starfield.prototype.stop = function () {
+  clearInterval(this.intervalId);
 };
 
-Starfield.prototype.update = function() {
-	var dt = 1 / this.fps;
-
-	for(var i=0; i<this.stars.length; i++) {
-		var star = this.stars[i];
-		star.y += dt * star.velocity;
-		//	If the star has moved from the bottom of the screen, spawn it at the top.
-		if(star.y > this.height) {
-			this.stars[i] = new Star(Math.random()*this.width, 0, Math.random()*3+1, 
-		 	(Math.random()*(this.maxVelocity - this.minVelocity))+this.minVelocity);
-		}
-	}
+Starfield.prototype.update = function () {
+  var dt = 1 / this.fps;
+
+  for (var i = 0; i < this.stars.length; i++) {
+    var star = this.stars[i];
+    star.t += dt;
+    //	If the star has moved from the bottom of the screen, spawn it at the top.
+    var x = star.x * star.t ** 2 * star.velocity;
+    var y = star.y * star.t ** 2 * star.velocity;
+    if (
+      x < -0.5 * this.width ||
+      x > 0.5 * this.width ||
+      y > 0.5 * this.height ||
+      y < -0.5 * this.height
+    ) {
+      var angle = 2 * Math.PI * Math.random();
+      var x = Math.cos(angle);
+      var y = Math.sin(angle);
+      var velocity =
+        Math.random() * (this.maxVelocity - this.minVelocity) +
+        this.minVelocity;
+      this.stars[i] = new Star(x, y, Math.random() * 3 + 1, velocity);
+    }
+  }
 };
 
-Starfield.prototype.draw = function() {
-
-	//	Get the drawing context.
-	var ctx = this.canvas.getContext("2d");
-
-	//	Draw the background.
- 	ctx.fillStyle = '#000000';
-	ctx.fillRect(0, 0, this.width, this.height);
-
-	//	Draw stars.
-	ctx.fillStyle = '#ffffff';
-	for(var i=0; i<this.stars.length;i++) {
-		var star = this.stars[i];
-		ctx.fillRect(star.x, star.y, star.size, star.size);
-	}
+Starfield.prototype.draw = function () {
+  //	Get the drawing context.
+  var ctx = this.canvas.getContext("2d");
+
+  //	Draw the background.
+  ctx.fillStyle = "#000000";
+  ctx.fillRect(0, 0, this.width, this.height);
+
+  //	Draw stars.
+  ctx.fillStyle = "#ffffff";
+  for (var i = 0; i < this.stars.length; i++) {
+    var star = this.stars[i];
+    var x = star.x * star.t ** 2 * star.velocity;
+    var y = star.y * star.t ** 2 * star.velocity;
+    var relX = Math.abs((2 * x) / this.width) ** 1.2;
+    var relY = Math.abs((2 * y) / this.height) ** 1.2;
+    ctx.fillStyle = `rgb(
+        ${Math.floor(relX * 255)},
+        ${Math.floor(relY * 255)},
```

Comment:
I would desaturate the colours so that they compete less with the text, which should remain visually  brighter/stronger.

```suggestion
        ${Math.floor(relX * 100)},
        ${Math.floor(relY * 100)},
```

I think my suggested code just makes it dark, maybe you'd have a better solution there.